editor/plugins/gdextension/cpp_scons/cpp_scons_gdext_creator.cpp

```diff
@@ -0,0 +1,232 @@
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "cpp_scons_gdext_creator.h"
+
+#include "core/core_bind.h"
+#include "core/io/dir_access.h"
+#include "core/version.h"
+#include "gdextension_template_files.gen.h"
+
+#include "editor/editor_node.h"
+
+void CppSconsGDExtensionCreator::_git_clone_godot_cpp(const String &p_parent_path, bool p_compile) {
+	EditorProgress ep("Preparing GDExtension C++ plugin", "Preparing GDExtension C++ plugin", 3);
+	ep.step("Checking for Git...", 0);
+	List<String> args;
+	args.push_back("--version");
+	String output = "";
+	int result = OS::get_singleton()->execute("git", args, &output);
+	ERR_FAIL_COND_MSG(result != 0 || output.is_empty(), "Could not run git command. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
+	args.front()->set("clone");
+	args.push_back("--single-branch");
+	args.push_back("--branch");
+	args.push_back(VERSION_BRANCH);
+	args.push_back("https://github.com/godotengine/godot-cpp");
+	const String godot_cpp_path = p_parent_path.trim_prefix("res://").path_join("godot-cpp");
+	args.push_back(godot_cpp_path);
+	ep.step("Cloning godot-cpp...", 1);
+	output = "";
+	result = OS::get_singleton()->execute("git", args, &output);
+	Ref<DirAccess> dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	if (result != 0 || !dir->dir_exists(godot_cpp_path)) {
+		args.get(3) = "master";
+		output = "";
+		result = OS::get_singleton()->execute("git", args, &output);
+	}
+	ERR_FAIL_COND_MSG(result != 0 || !dir->dir_exists(godot_cpp_path), "Failed to clone godot-cpp. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
```

Comment:
If cloning manually even an option? You are not checking whether it exists before trying to clone.

Reply:
The user would need to either install Git first, or just download the ZIP from the website. The lack of Git at this point in the code does not mean cloning manually is impossible to do later.

Reply:
What I mean is that you made it basically a required step, so the dialog can't be used with manually cloned repo, and the error message somewhat suggests otherwise.

Reply:
I don't understand what you mean. You can use the dialog with a manually cloned repo just fine? The error message is in fact not suggesting otherwise, it is saying that you *should* manually clone the repo?